src/Microsoft.AspNet.Mvc.Core/ApplicationModels/ActionModel.cs

```diff
@@ -48,7 +48,7 @@ public ActionModel([NotNull] ActionModel other)
             }
         }
 
-        public List<IActionConstraintMetadata> ActionConstraints { get; private set; }
+        public IList<IActionConstraintMetadata> ActionConstraints { get; private set; }
```

Comment:
Unrelated, but we should get around to documenting these properties particularly since we want users to be using these types more often now.